src/init/resolvers.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * AR.IO Gateway
+ * Copyright (C) 2022-2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Logger } from 'winston';
+import { StandaloneArNSResolver } from '../resolution/standalone-arns-resolver.js';
+import { TrustedGatewayArNSResolver } from '../resolution/trusted-gateway-arns-resolver.js';
+import { NameResolver } from '../types.js';
+
+export const createArNSResolver = ({
+  log,
+  type,
+  url,
+}: {
+  log: Logger;
+  type: string;
```

Comment:
can we have a better type like `'resolver' | 'gateway'`?

Reply:
bc it's provided as an environment variable - we'd have to force the type using `as` when calling the function [here](https://github.com/ar-io/ar-io-node/pull/99/files#diff-6fe6a66d9e19964283ad8fcf5ad1a9bf0e8a32a22124ef6f28474d92fda574edR395-R396) - which makes the type somewhat useless - the `switch` statement handles invalid values appropriately but at runtime.

TLDR - we could set this to `'resolver' | 'gateway' | string` to avoid forcing types on env vars